samza-core/src/main/scala/org/apache/samza/container/RunLoopTask.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.container
+
+import org.apache.samza.checkpoint.OffsetManager
+import org.apache.samza.scheduler.EpochTimeScheduler
+import org.apache.samza.system.{IncomingMessageEnvelope, SystemStreamPartition}
+import org.apache.samza.task.{ReadableCoordinator, TaskCallbackFactory}
+import org.apache.samza.util.Logging
+
+abstract class RunLoopTask extends Logging {
```

Comment:
The extension of `Logging` is a little bit of a problem. We need to either: convert `Logging` to java as well, or (and this is according to my limited knowledge of java <--> scala cross functionality), because the trait has members, create a new scala *class* which extends `Logging` and extend that from the abstract class java version of RunLoopTask. If we do the former I think it should be kept outside the scope of this PR. I'm open to the latter though, just think it would be a bit odd to have an additional scala class with no body for the purpose of facilitating cross-functionality with java.

What are your thoughts?